lib/stm32/common/flash_common_f24.c

```diff
@@ -332,10 +333,27 @@ was not properly erased.
 
 void flash_program(uint32_t address, const uint8_t *data, uint32_t len)
 {
-	/* TODO: Use dword and word size program operations where possible for
-	 * turbo speed.
-	 */
 	uint32_t i;
+
+	if ((address & 0x3) != 0) {
+		uint32_t to_align = 4 - (address & 0x3);
+		for (i = 0; i < to_align; i++) {
+			flash_program_byte(address, *data);
+			address++;
+			data++;
+		}
+		len -= to_align;
+	}
+
+	while (len >= 4) {
+		uint32_t d;
+		memcpy(&d, data, sizeof(d));
```

Comment:
ah !! direct memcpy to a uint32 !!! I know, stm32* all have the same endianness and there is 0 portability issue here. But I do believe it would be preferable to replace that memcpy with a simple `d = (data[3] << 24) | (data[2] << 16) [...]` construct. One less call to memcpy.

Reply:
fixed

Reply:
perfect ! tested too I assume ? (I haven't)

Reply:
The version of the code using `memcpy` has been used extensively on a bunch of stm32f427. I didn't test the version without `memcpy` as I no longer have the hardware to do so, but it compiles and I don't see how it could be wrong :)